src/main/java/org/gridsuite/study/server/service/StudyService.java

```diff
@@ -608,8 +608,12 @@ public String getLineMapData(UUID studyUuid, UUID nodeUuid, String lineId, boole
                 "lines", lineId);
     }
 
-    public String getTwoWindingsTransformersMapData(UUID studyUuid, UUID nodeUuid, List<String> substationsIds) {
-        return networkMapService.getEquipmentsMapData(networkStoreService.getNetworkUuid(studyUuid), networkModificationTreeService.getVariantId(nodeUuid),
+    public String getTwoWindingsTransformersMapData(UUID studyUuid, UUID nodeUuid, List<String> substationsIds, boolean inUpstreamBuiltParentNode) {
+        UUID nodeUuidToSearchIn = nodeUuid;
```

Comment:
UUID nodeUuidToSearchIn = nodeUuid;
        if (inUpstreamBuiltParentNode) {
            nodeUuidToSearchIn = networkModificationTreeService.doGetLastParentNodeBuilt(nodeUuid);
        }

This code could be mutualized nope ?

Reply:
Yes
method `getNodeUuidFrom(inUpstreamBuiltParentNode)` ?

Reply:
Created a method getNodeUuidToSearchIn